Doc/whatsnew/3.10.rst

```diff
@@ -135,17 +135,35 @@ Porting to Python 3.10
 
 * Since :c:func:`Py_TYPE()` is changed to the inline static function,
   ``Py_TYPE(obj) = new_type`` must be replaced with ``Py_SET_TYPE(obj, new_type)``:
-  see :c:func:`Py_SET_TYPE()` (available since Python 3.9).
+  see :c:func:`Py_SET_TYPE()` (available since Python 3.9). For backward
+  compatibility, this macro can be used::
+
+      #if PY_VERSION_HEX < 0x030900A4
+      #  define Py_SET_TYPE(obj, type) do { Py_TYPE(obj) = (type); } while (0)
+      #endif
+
   (Contributed by Dong-hee Na in :issue:`39573`.)
 
 * Since :c:func:`Py_REFCNT()` is changed to the inline static function,
   ``Py_REFCNT(obj) = new_refcnt`` must be replaced with ``Py_SET_REFCNT(obj, new_refcnt)``:
-  see :c:func:`Py_SET_REFCNT()` (available since Python 3.9).
+  see :c:func:`Py_SET_REFCNT()` (available since Python 3.9). For backward
+  compatibility, this macro can be used::
+
+      #if PY_VERSION_HEX < 0x030900A4
+      #  define Py_SET_REFCNT(obj, refcnt) do { Py_REFCNT(obj) = (refcnt); } while (0)
```

Comment:
@tacaswell: In one of your PR, you wrote "typ" instead of "type". Was it a deliberate typo? Is there an issue in C++? Should I rename the variable to "new_type"?

Reply:
Oops, my question is on the second parameter of Py_SET_TYPE() :-)

Reply:
I built Py_SET_TYPE() macro with g++ and Python 3.8: I don't have any warning or error. "type" sounds like a valid parameter type.

Reply:
I am not sure, that code come from @eric-wieser via https://github.com/numpy/numpy/pull/16417#issuecomment-635491783